pkg/accounting/accounting.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package accounting
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var _ Interface = (*Accounting)(nil)
+
+// Interface is the main interface for Accounting
+type Interface interface {
+	// Reserve reserves a portion of the balance for peer
+	// It returns an error if the operation would risk exceeding the disconnect threshold
+	// This should be called (always in combination with release) before a Credit action
+	Reserve(peer swarm.Address, price uint64) error
+	// Release releases reserved funds
+	Release(peer swarm.Address, price uint64)
+	// Credit increases the balance the peer has with us (we "pay" the peer)
+	Credit(peer swarm.Address, price uint64) error
+	// Debit increases the balance we have with the peer (we get "paid")
+	Debit(peer swarm.Address, price uint64) error
+	// Balance returns the current balance for the given peer
+	Balance(peer swarm.Address) (int64, error)
```

Comment:
Please expose this in the debug API

Reply:
I would keep the api to get this (and also the api to get all balances) out of scope of this PR.

Reply:
Ok. Good for me. The reason why I think it would be good to have it in is that this would make it more complete, from the user perspective. Now, a lot of things are happening in the background, but there is no way to the user to find out what exactly has happened.

Reply:
But doing it in upcoming PRs is ok, of course!

Reply:
Ok